src/Datadog.Trace/TracingProcessManager.cs

```diff
@@ -390,32 +400,38 @@ private void OnPortFileDeleted(object source, FileSystemEventArgs e)
 
             private void ReadPortAndAlertSubscribers()
             {
-                try
+                var retries = 3;
+
+                while (retries-- > 0)
                 {
-                    var portFile = PortFilePath;
-                    var portText = ReadSingleLineNoLock(portFile);
-                    if (int.TryParse(portText, NumberStyles.Any, CultureInfo.InvariantCulture, out var portValue))
+                    try
                     {
-                        if (Port == portValue)
+                        var portFile = PortFilePath;
+                        var portText = ReadSingleLineNoLock(portFile);
+                        if (int.TryParse(portText, NumberStyles.Any, CultureInfo.InvariantCulture, out var portValue))
+                        {
+                            if (Port == portValue)
+                            {
+                                // nothing to do, let's not cause churn
+                                return;
+                            }
+
+                            Port = portValue;
+                            DatadogLogging.RegisterStartupLog(log => log.Debug("Retrieved port {0} from {1}.", portValue, PortFilePath));
+                            RefreshPortVars();
+                        }
+                        else
                         {
-                            // nothing to do, let's not cause churn
-                            return;
+                            DatadogLogging.RegisterStartupLog(log => log.Error("The port file ({0}) is malformed: {1}", PortFilePath, portText));
                         }
 
-                        Port = portValue;
-                        DatadogLogging.RegisterStartupLog(log => log.Debug("Retrieved port {0} from {1}.", portValue, PortFilePath));
-                        RefreshPortVars();
+                        AlertSubscribers();
                     }
-                    else
+                    catch (Exception ex)
                     {
-                        DatadogLogging.RegisterStartupLog(log => log.Error("The port file ({0}) is malformed: {1}", PortFilePath, portText));
+                        DatadogLogging.RegisterStartupLog(log => log.Error(ex, "Error when alerting subscribers for {0}", Name));
+                        Thread.Sleep(5); // Wait just a tiny bit just to let the file come unlocked
```

Comment:
It seems like you're trying to implement a retry logic of sorts in here. A lot of code going into this. I wonder if there's a simpler retry-logic that could be applied. Too bad we don't have Polly incorporated in the tracer, it might be useful here. But that's a bigger change. Just some thoughts.

Reply:
I do think we should have a "RetryHelper" that takes a delegate and some configuration around number of times to retry and delays.
I've implemented many retries in my life.

Reply:
It would get rid of a lot of duplicated code